tests/utils.go

```diff
@@ -237,12 +237,29 @@ type ProcessFunc func(event *k8sv1.Event) (done bool)
 type ObjectEventWatcher struct {
 	object                 runtime.Object
 	timeout                *time.Duration
-	failOnWarnings         bool
 	resourceVersion        string
 	startType              startType
+	warningPolicy          WarningsPolicy
 	dontFailOnMissingEvent bool
 }
 
+type WarningsPolicy struct {
+	FailOnWarnings     bool
+	WarningsIgnoreList []string
+}
+
+func (wp *WarningsPolicy) shouldIgnoreWarning(event *k8sv1.Event) bool {
```

Comment:
@oshoval I think the pointer receiver here is a little bit confusing because the function doesn't modify the struct the receiver is pointing to.

Reply:
as we spoke
You may want to use a pointer receiver type to avoid copying on method calls or to allow the method to mutate the receiving struct.
https://gobyexample.com/methods

Reply:
@oshoval Agree. Thank you!